docs/design-docs/security-groups/SECURITY-GROUPS.md

```diff
@@ -0,0 +1,364 @@
+- Whenever we want to add new rule - for example open port "X", we should *COPY* all current roles into our deployment .yaml file, and at the end, add the rule which we want.
+- Each component has his own rule-set, so we need to be very carefull where we're putting them.
+- After coping, we can also modify existing default security groups.
+- After adding new rules, and infra part is done (terraform), we can go into terraform build directory and check if fiiles contain our port definition.
+
+## Security groups diagram
+Check bellow security diagram, which show how security groups are related to other components. This is example of AWS architecutre, but in Azure should be almost the same.
+![](../AWS/aws_cluster_setup.svg)
+
+## Azure Security groups
+List of all security groups and related services in Azure  are described [here](https://github.com/epiphany-platform/epiphany/blob/develop/core/src/epicli/data/azure/defaults/infrastructure/virtual-machine.yml).
+
+
+Rules description:
+```yaml
+- name:                       "Name of the rule"
+  description:                "Short rule description"
+  priority:                   "Priority (NUM), which describes which rules should be taken into conediration as first "
+  direction:                  "Inbound || Outbound" - which direction are you allowing rule"
+  access:                     "Allow|Deny - whenever we want to grant access or block"
+  protocol:                   "TCP || UDP" - which protocol should be used for connections"
+  source_port_range:          "Source port ranges"
+  destination_port_range:     "Destination port/s range"
+  source_address_prefix:      "Source network address"
+  destination_address_prefix: "Destination network address"
+```
+
+Lets look into example on which, we are setting new rule name "nrpe-agent-port", with priority 250, which is allowing accesses from local network "10.1.4.0/24" into all hosts in our network on port 5666.
+
+The rule:
+```yaml
+     - name: nrpe-agent-port
+       description: Allow access all hosts on port 5666 where nagios agent is running.
+       priority: 250
+       direction: Inbound
+       access: Allow
+       protocol: Tcp
+       source_port_range: "*"
+       destination_port_range: "5666"
+       source_address_prefix: "10.1.4.0/24"
+       destination_address_prefix: "0.0.0.0/0"
+```
+
+## Azure Security groups full yaml file
+To deploy previously mentioned rule, we need to setup a complete YAML configuraiton file. Bellow example shows how this file should looks like. In this configuration we set simple setup of epiphany with 2nodes and 1 master vm in Azure.
+```yaml
+kind: epiphany-cluster
+name: default
+provider: azure
+title: Epiphany cluster Config
+specification:
+  name: azure
+  prefix: azure
+  admin_user:
+    name: operations
+    key_path:  /workspaces/epiphany/core/src/epicli/clusters/keys/abb_rsa
+  cloud:
```

Comment:
It could be nice to add the latest change in epiphany config file for v0.8.0
```
      cloud:
        k8s_as_cloud_service: false # <----- make sure that flag is set, as it indicates usage of AKS service
        .... 
```

Reply:
Default is false so don't think its needed for the this.